docs/source/deploying_services_advanced.rst

```diff
@@ -0,0 +1,48 @@
+.. _deploying_services_advanced:
+
+===========================================
+How to deploy a service (developer's guide)
+===========================================
+This is a guide for developers that want to deploy Octue services themselves - it is not needed if Octue manages your
+services for you or if you are only asking questions to existing Octue services.
+
+1. Ensuring you are in the desired project, go to the `Google Cloud Run <https://console.cloud.google.com/run>`_ page
+   and create a new service
+2. Give your service a unique and useful name
+3. Choose a low-carbon region that supports Eventarc triggers and is in a convenient geographic location for you (e.g.
+   physically close to you for low latency or in a region compatible with your data protection requirements)
+3. Click "Next" and select "Continuously deploy new revisions from a source repository", then click "Set up with cloud
+   build".
+4. Choose your source code repository provider and the repository containing the code you'd like to deploy. You'll have
+   to give the provider permission to access the repository. If your provider isn't GitHub, BitBucket, or Google Cloud
+   Source Repositories (GCSR), you'll need to mirror the repository to GCSR before completing this step.
+5. Click "Next", enter a regular expression for the branches you want to automatically deploy from (``main`` by default),
+   select "Dockerfile", and click "Save".
+6. Click "Next", select "Allow internal traffic only" and "Require authentication", then click "Add eventarc trigger".
+7. Choose "Cloud Pub/Sub topic" as the trigger event, click on the menu called "Select a Cloud Pub/Sub topic", then
```

Comment:
Add some explanations of why you’re doing things.

for example: “When changes are made to the source code, we want them to be deployed automatically. So, we need to connect the repository to GCP to enable this…”